lib/queryHooks.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+const async = require('async');
+const executeHook = require('./executeHook');
+const checkHooks = require('./logHooks');
+
+module.exports = (server, settings, collection, allDone) => {
+  async.autoInject({
+    outstandingHooks(done) {
+      checkHooks(collection, done);
+    },
+    logHooks(outstandingHooks, done) {
+      server.log(['hapi-hooks', 'status'], outstandingHooks);
+      // does logging statement, checks if need to continue:
+      if (outstandingHooks.processing !== 0) {
+        server.log(['hapi-hooks', 'busy'], `There are still ${outstandingHooks.processing} hooks in the queue.`);
+        return allDone();
```

Comment:
It's a little confusing to call allDone inside the auto. Let's return done with an error